spmd/tensor/api.py

```diff
@@ -304,6 +304,22 @@ def to_local(self) -> torch.Tensor:
         """
         return ToTorchTensor.apply(self)  # pyre-ignore[16]: autograd func
 
+    def to_logical_tensor(self) -> torch.Tensor:
```

Comment:
why do we need an API for this?

Reply:
This API was intended mainly for testing usage, because there're a lot of boilerplate code that just transforms a DTensor to replication then do the comparison. I was thinking about this more and I think it might not make sense to introduce this API yet, so I instead add a helper function in DistTensorTestBase to make testing easier.